README.md

```diff
@@ -1,3 +1,134 @@
+
+__Serverless:__ The Serverless framework is utilized to deploy the actual application that will handle the event source data. This framework creates the following resources:
+ - A custom authorizer for webhook based APIs that use a header token.
+ - An API Gateway for receiving post events using the authorizer.
+ - Custom configuration variables that are populated by the `buildspec.yml` located in each event source configuration's CodeCommit repository.
+ - A lambda event handler that will validate to some degree the event to be processed by MozDef. At some point we may extend functionality here to fully process the event by another lambda.
+ - An SQS queue to receive the events that have been handled by our lambda handler.
+ - A Dead Letter Queue (DLQ) to receive events that fail to be handled, so that they can be reprocessed once a fix is introduced. This is future functionality that has not been implemented yet.
+
+We plan to include testing modules in this framework as well.
+
+To deploy this framework you will need the following established:
+
+1. You'll need to design your parameter store pathing, and add your tokens or other data to be used with the framework. We used the following structure:
+
+`/<Project_name>/<event_source_name>/<environemnt>/<authorizer_token_var>/`
+
+This allows us to keep track of the various event sources, what environment they are used in, and keep the same variables across all event sources that will contain different values.
+
+2. An AWS Account with the ability to use all of the AWS services previously mentioned.
+
+3. A CodeCommit repository that contains your buildspec and deploy script.
+
+Please note the env: variables that must be filled out, these can essentially be whatever you want them to be.
+___
+
+__Example of a Buildspec:__
+
+```
+    version: 0.2 
+    
+    env:  
+    variables: 
+        STAGE: <dev|stage|prod> 
+        SERVICE: <event_source_service>
+        PROJECT: <project_name> 
+        API_PATH: <api_path_to_post_to> 
+    
+    phases: 
+    install: 
+        runtime-versions: 
+        python: 3.7 
+        nodejs: 10 
+        commands: 
+        # Install dependencies here 
+        - pip3 install --upgrade awscli -q 
+        - npm install -g --silent --progress=false serverless 
+        - npm install --silent --save-dev serverless-pseudo-parameters 
+        - npm install --silent --save-dev serverless-prune-plugin 
+    pre_build: 
+        commands: 
+        # Perform pre-build actions here, maybe tests for instance 
+    build: 
+        commands: 
+        # Invoke the deploy script here 
+        - chmod +x $CODEBUILD_SRC_DIR/config/deploy.sh 
+        - $CODEBUILD_SRC_DIR/config/deploy.sh deploy $STAGE $AWS_REGION 
+```
```

Comment:
The indentation appears to be off here, maybe it's just Markdown

Reply:
Fixed!